### PR TITLE
1.16 updates

### PR DESCRIPTION
**Pull request by kickash32**
Compiles,
Not tested yet!

### CODE REVIEW

**Comment by Puremin0rez** [comment]
Armor stand ticking patch wasn't added back in yet (at least in current ver/1.16 branch), shouldn't be dropped from the /removed/ folder

File: Spigot-Server-Patches/0521-Limit-lightning-strike-effect-distance.patch [hunk 1]
```diff
@@ -1,17 +1,43 @@
 From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
-From: Trigary <trigary0@gmail.com>
-Date: Fri, 14 Sep 2018 17:42:08 +0200
-Subject: [PATCH] Limit lightning strike effect distance
+From: Aikar <aikar@aikar.co>
+Date: Mon, 17 Sep 2018 23:05:31 -0400
+Subject: [PATCH] Support Overriding World Seeds
```
Comment (Proximyst) [inline Spigot-Server-Patches/0521-Limit-lightning-strike-effect-distance.patch]: Seems you mixed two patches somehow?

File: Spigot-Server-Patches/0521-Limit-lightning-strike-effect-distance.patch [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
 diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
-index 487b0d5cd608e84a793eba5fdbd50a9f3d95c79b..b8789c8ecc5a6e4117bb7ce0d5487a6e5774b67f 100644
+index 0f4fca90fd6c3788a5762c96c344899cb1665466..4a59d42b778e681e93f1243ecd16c5cf46160b97 100644
 --- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
 +++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
-@@ -242,6 +242,28 @@ public class PaperWorldConfig {
+@@ -1,11 +1,5 @@
+ package com.destroystokyo.paper;
+ 
+-import java.util.Arrays;
```
Comment (Proximyst) [inline Spigot-Server-Patches/0521-Limit-lightning-strike-effect-distance.patch]: Unnecessary diff (just moved down).

File: Spigot-Server-Patches/0522-Add-permission-for-command-blocks.patch [hunk 3]
```diff
@@ -5,58 +5,58 @@ Subject: [PATCH] Add permission for command blocks
...
-@@ -110,7 +110,7 @@ public class BlockCommand extends BlockTileEntity {
+@@ -105,7 +105,7 @@ public class BlockCommand extends BlockTileEntity {
      public EnumInteractionResult interact(IBlockData iblockdata, World world, BlockPosition blockposition, EntityHuman entityhuman, EnumHand enumhand, MovingObjectPositionBlock movingobjectpositionblock) {
          TileEntity tileentity = world.getTileEntity(blockposition);
  
 -        if (tileentity instanceof TileEntityCommand && entityhuman.isCreativeAndOp()) {
 +        if (tileentity instanceof TileEntityCommand && (entityhuman.isCreativeAndOp() || (entityhuman.isCreative() && entityhuman.getBukkitEntity().hasPermission("minecraft.commandblock")))) { // Paper - command block permission
              entityhuman.a((TileEntityCommand) tileentity);
-             return EnumInteractionResult.SUCCESS;
+             return EnumInteractionResult.a(world.isClientSide);
```
Comment (Proximyst) [inline Spigot-Server-Patches/0522-Add-permission-for-command-blocks.patch:18]: This seems like something that would benefit from an obf helper.

File: Spigot-Server-Patches/0523-Ensure-Entity-AABB-s-are-never-invalid.patch [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+ package net.minecraft.server;
+ 
++import co.aikar.timings.MinecraftTimings;
++import co.aikar.timings.Timing;
+ import com.google.common.collect.Iterables;
+ import com.google.common.collect.Lists;
+ import com.google.common.collect.Sets;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleArrayMap;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleMap;
+-import java.util.Arrays;
```
Comment (Proximyst) [inline Spigot-Server-Patches/0523-Ensure-Entity-AABB-s-are-never-invalid.patch:21]: Again, moving down.

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0522-Add-permission-for-command-blocks.patch (hunk 3 above)
Comment (kickash32) [reply Spigot-Server-Patches/0522-Add-permission-for-command-blocks.patch:18]: we don't touch those lines, its irrelevent
